File: app/src/main/java/com/example/firefighters/models/EmergencyModel.java
package com.example.firefighters.models;

import com.example.firefighters.tools.ConstantsValues;

import java.util.Calendar;

public class EmergencyModel {
    private String id;

    private String senderMail; //Foreign key user sender
    private String messageId;  //Foreign key message

    private int sendUtc;
    private double longitude;
    private double latitude;
    private String streetName;
    private String gravity;
    private String status;
    private String sendDate;
    private String sendHour;

    public EmergencyModel() {
        gravity = ConstantsValues.GRAVITY_NORMAL;
        status = ConstantsValues.NOT_WORKING;

        Calendar calendar = Calendar.getInstance();
        int year = calendar.get(Calendar.YEAR);
        int month = calendar.get(Calendar.MONTH);
        int day = calendar.get(Calendar.DAY_OF_MONTH) + 1;
        int utc = calendar.get(Calendar.ZONE_OFFSET);
        int hours = calendar.get(Calendar.HOUR_OF_DAY);
        int minutes = calendar.get(Calendar.MINUTE);
        int sec = calendar.get(Calendar.SECOND);
        sendDate = day + "/" + month + "/" + year;
        sendHour = hours + ":" + minutes + ":" + sec;
        sendUtc = utc;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getSenderMail() {
        return senderMail;
    }

    public void setSenderMail(String senderMail) {
        this.senderMail = senderMail;
    }

    public String getMessageId() {
        return messageId;
    }

    public void setMessageId(String messageId) {
        this.messageId = messageId;
    }

    public int getSendUtc() {
        return sendUtc;
    }

    public void setSendUtc(int sendUtc) {
        this.sendUtc = sendUtc;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public String getStreetName() {
        return streetName;
    }

    public void setStreetName(String streetName) {
        this.streetName = streetName;
    }

    public String getGravity() {
        return gravity;
    }

    public void setGravity(String gravity) {
        this.gravity = gravity;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getSendDate() {
        return sendDate;
    }

    public void setSendDate(String sendDate) {
        this.sendDate = sendDate;
    }

    public String getSendHour() {
        return sendHour;
    }

    public void setSendHour(String sendHour) {
        this.sendHour = sendHour;
    }

}

File: app/src/main/java/com/example/firefighters/models/EmergencyModel.kt
package com.example.firefighters.models

import com.example.firefighters.utils.ConstantsValues
import java.util.Calendar

class EmergencyModel {
    var id: Long = 0
    var senderMail: String? = null //Foreign key user sender
    var messageId: Int = 0 //Foreign key message
    var sendUtc: Int
    var longitude: Double = 0.0
    var latitude: Double = 0.0
    var gravity: String
    var status: String
    var sendDate: String
    var sendHour: String
    var currentUnit: String? = null

    init {
        gravity = ConstantsValues.GRAVITY_NORMAL
        status = ConstantsValues.NOT_WORKING
        val calendar = Calendar.getInstance()
        val year = calendar[Calendar.YEAR]
        val month = calendar[Calendar.MONTH]
        val day = calendar[Calendar.DAY_OF_MONTH] + 1
        val utc = calendar[Calendar.ZONE_OFFSET]
        val hours = calendar[Calendar.HOUR_OF_DAY]
        val minutes = calendar[Calendar.MINUTE]
        val sec = calendar[Calendar.SECOND]
        sendDate = "$day/$month/$year"
        sendHour = "$hours:$minutes:$sec"
        sendUtc = utc
    }
}
